subsys/net/lib/nrf_cloud/src/nrf_cloud_client_id.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_INTERNAL_UUID)
+#include <modem/modem_jwt.h>
+#endif
+#include <modem/at_cmd.h>
+#include <zephyr.h>
+#include <stdio.h>
+#include <logging/log.h>
+#include "nrf_cloud_transport.h"
+
+LOG_MODULE_REGISTER(nrf_cloud_client_id, CONFIG_NRF_CLOUD_LOG_LEVEL);
+
+#if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_COMPILE_TIME)
+BUILD_ASSERT((sizeof(CONFIG_NRF_CLOUD_CLIENT_ID) - 1) <= NRF_CLOUD_CLIENT_ID_MAX_LEN,
+	"CONFIG_NRF_CLOUD_CLIENT_ID must not exceed NRF_CLOUD_CLIENT_ID_MAX_LEN");
+#endif
+
+#if defined(CONFIG_NRF_CLOUD_CLIENT_ID_SRC_IMEI)
+#define CGSN_RESPONSE_LENGTH 19
+#define NRF_IMEI_LEN 15
+#define IMEI_CLIENT_ID_LEN (sizeof(CONFIG_NRF_CLOUD_CLIENT_ID_PREFIX) \
+			    - 1 + NRF_IMEI_LEN)
+BUILD_ASSERT(IMEI_CLIENT_ID_LEN <= NRF_CLOUD_CLIENT_ID_MAX_LEN,
+	"NRF_CLOUD_CLIENT_ID_PREFIX plus IMEI must not exceed NRF_CLOUD_CLIENT_ID_MAX_LEN");
+#endif
+
+int nrf_cloud_client_id_get(char *id_buf, size_t id_len)
+{
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+	return nct_client_id_get(id_buf, id_len);
+#else
+	return nrf_cloud_cfgd_client_id_get(id_buf, id_len);
```

Comment:
Just out of curiosity, what does "cfgd" stand for? Configured? :)

Reply:
yes.
shall I just use the whole word instead of cfgd?

Reply:
Please do, that would be clearer.

Reply:
done